rules/TYPO312/v0/MigrateBackendModuleRegistrationRector.php

```diff
@@ -0,0 +1,499 @@
+    {
+        try {
+            $this->parseFileNodes($file);
+        } catch (ShouldNotHappenException $shouldNotHappenException) {
+            throw $shouldNotHappenException;
+        } catch (AnalysedCodeException $analysedCodeException) {
+            return new SystemError($analysedCodeException->getMessage(), $file->getFilePath(), $analysedCodeException->getLine());
+        } catch (Throwable $throwable) {
+            return new SystemError($throwable->getMessage(), $file->getFilePath(), $throwable->getLine());
+        }
+
+        return null;
+    }
+
+    private function parseFileNodes(File $file): void
+    {
+        // store tokens by original file content, so we don't have to print them right now
+        $stmtsAndTokens = $this->rectorParser->parseFileContentToStmtsAndTokens($file->getOriginalFileContent());
+        $oldStmts = $stmtsAndTokens->getStmts();
+        $oldTokens = $stmtsAndTokens->getTokens();
+        $newStmts = $this->nodeScopeAndMetadataDecorator->decorateNodesFromFile($file->getFilePath(), $oldStmts);
+        $file->hydrateStmtsAndTokens($newStmts, $oldStmts, $oldTokens);
+    }
+
+    private function mergeFiles_(File $existingFile, File $newFile): void
+    {
+        $existingStmts = $existingFile->getNewStmts();
+
+        /** @var FileWithoutNamespace $fileWithoutNamespace */
+        $fileWithoutNamespace = $existingStmts[0];
+
+        $existingArray = $this->getNodeArray($fileWithoutNamespace);
+
+        // --- new php array
+
+        $newStmts = $newFile->getNewStmts();
+
+        /** @var FileWithoutNamespace $fileWithoutNamespace2 */
+        $fileWithoutNamespace2 = $newStmts[0];
+
+        $newArray = $this->getNodeArray($fileWithoutNamespace2);
+
+        // Merge the two arrays
+        $existingArray->items[] = $newArray->items[0];
+
+        $existingFile->changeNewStmts($existingStmts);
+    }
+
+    private function mergeFiles(File $existingFile, ArrayItem $arrayItem): void
+    {
+        $existingStmts = $existingFile->getNewStmts();
+
+        /** @var FileWithoutNamespace $fileWithoutNamespace */
+        $fileWithoutNamespace = $existingStmts[0];
+
+        $existingArray = $this->getNodeArray($fileWithoutNamespace);
+
+        // --- new php array
+
+        // Merge the two arrays
+        $existingArray->items[] = $arrayItem;
+
+        $existingFile->changeNewStmts($existingStmts);
+    }
+
+    private function getNodeArray(FileWithoutNamespace $fileWithoutNamespace): Array_
+    {
+        /** @var Return_ $return */
+        $return = $fileWithoutNamespace->stmts[0];
+
+        /** @var Array_ $array */
+        $array = $return->expr;
+
+        return $array;
+    }
+
+    private function printFile(File $file, string $filePath) : void
+    {
+        // only save to string first, no need to print to file when not needed
+        $newContent = $this->formatPreservingPrinter->printParsedStmstAndTokensToString($file);
```

Comment:
this removed at 

- https://github.com/rectorphp/rector-src/pull/6036

use directly from `betterStandardPrinter` instead:

```php
        $newContent = $this->betterStandardPrinter->printFormatPreserving(
             $file->getNewStmts(),
            $file->getOldStmts(),
            $file->getOldTokens()
       );
```